tests/Functional/ChangeLinkTest.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Test\Functional;
+
+use App\Entity\Link;
+use App\Repository\LinkRepositoryInterface;
+use DI\Container;
+
+class ChangeLinkTest extends FeatureTestCase
+{
+    /**
+     * @var LinkRepositoryInterface|mixed|\PHPUnit\Framework\MockObject\MockObject
+     */
+    private mixed $repository;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        /** @var Container $container */
+        $container = $this->getApp()->getContainer();
+
+        $container->set(
+            LinkRepositoryInterface::class,
+            $this->repository = $this->createMock(LinkRepositoryInterface::class)
+        );
+    }
+
+    public function testSuccess(): void
+    {
+        $this->repository
+            ->method('get')
+            ->with('abc3')
+            ->willReturn(new Link(
+                'abc3',
+                'https://yandex.ru',
+                null,
+                []
+            ))
+        ;
+
+        $this->repository
+            ->expects($this->once())
+            ->method('save')
+            ->with(new Link(
+                'abc3',
+                'https://google.com',
+                'Cool link to google',
+                [
+                    'homepage',
+                    'mylink'
+                ]
+            ))
+        ;
+
+        $body = [
+            'long_url' =>'https://google.com',
+            'title' => 'Cool link to google',
+            'tags' =>  ['homepage', 'mylink'],
+        ];
+
+        $response = $this->jsonPatch('/links/abc3', $body);
+
+        $this->assertEquals(204, $response->getStatusCode());
```

Comment:
По поводу тестов сказать ничего не могу. Тесты как тесты. кроме вот этого момента смотри
```suggestion
        self::assertEquals(204, $response->getStatusCode());
```
Этот метод лучше вызывать вот так. Тот же Psalm тебя бы тут поправил, но впринципе и Stan должен был, при правильной настройке.
Такое тут в тестах везде по мелочи.